queries.sql

```diff
@@ -73,3 +73,110 @@ FROM animals
+    FROM species
+    JOIN animals
+    on species.id = animals.species_id 
+    GROUP By species.name;
+ 
+SELECT full_name AS "OWNER NAME" , animals.name AS "ANIMALS NAME",species.name AS "Animal Species"
+   FROM animals
+   INNER JOIN owners ON  animals.owner_id = owners.id
+   INNER JOIN species ON  animals.species_id = species.id
+   WHERE owners.id = 2 and species.name = 'Digimon'; 
+
+ SELECT full_name AS "OWNER NAME" , animals.name AS "ANIMALS NAME",species.name AS "Animal Species"
+   FROM animals
+   INNER JOIN owners ON  animals.owner_id = owners.id
+   INNER JOIN species ON  animals.species_id = species.id
+   WHERE owners.id = 5 and animals.escape_attempts = 0; 
+
+   SELECT full_name AS "OWNER NAME",  COUNT(animals.name) AS "ANIMAL COUNT" 
+    FROM owners
+    JOIN animals
+    on owners.id = animals.owner_id 
+    GROUP By full_name
+    ORDER BY COUNT(animals.name) DESC LIMIT 1;
+
+ --  querys for many-to-many
+
+   SELECT animals.id, animals.name AS Last_Seen, visit_date
+   FROM animals JOIN visits 
+   ON animals.id = visits.animal_id
+   WHERE vets_id = (SELECT id from vets WHERE name = 'William Tatcher')
+   ORDER BY visit_date DESC LIMIT 1;
+
+    SELECT vets.name AS "VET NAME" , COUNT(animals.name) AS "animals count"
+    from animals
+    JOIN visits ON animals.id = visits.animal_id
+    JOIN vets ON visits.vets_id = vets.id
+    WHERE vets_id = (SELECT id from vets WHERE name = 'Stephanie Mendez')
+    GROUP BY vets.name;
+
+SELECT vets.id, vets.name AS "Vet name", species.name AS "SpecializATION"
+   FROM vets
+   LEFT JOIN specializations ON specializations.vets_id = vets.id
+   LEFT JOIN species on specializations.species_id = species.id;    
+       
+
+   SELECT vets.name AS "VET NAME" , animals.name AS "Animal Seen", visit_date
+    from animals
+    JOIN visits ON animals.id = visits.animal_id
+    JOIN vets ON visits.vets_id = vets.id
+    WHERE vets_id = (SELECT id from vets WHERE name = 'Stephanie Mendez') AND visit_date BETWEEN '2020-04-01' and '2020-08-30';
+
+ 
+  SELECT animals.name , COUNT (*) as "Visit count"
+   from animals
+   JOIN visits ON animals.id = visits.animal_id
+   GROUP BY  animals.name
+   ORDER BY COUNT (*) DESC LIMIT 1;
+
+  SELECT animals.name AS "FIRST ANIMAL SEEN", visit_date
+    from animals
+    JOIN visits
+    ON animals.id = visits.animal_id
+    WHERE vets_id = (SELECT id from vets WHERE name = 'Maisy Smith')
+    ORDER BY visit_date ASC LIMIT 1;
+
+SELECT animals.name AS "RECENT ANIMAL SEEN", vets.name AS "VET NAME", visit_date AS "Visit Date"
+    FROM animals
+    JOIN visits ON animals.id = visits.animal_id
+    JOIN vets ON visits.vets_id = vets.id
+    ORDER BY visit_date DESC LIMIT 1;
+
+SELECT COUNT(*) AS "number of visits" FROM vets
+    INNER JOIN visits ON vets.id = visits.vets_id
+    LEFT JOIN specializations ON vets.id = specializations.vets_id
+    LEFT JOIN species ON specializations.species_id = species.id
+    WHERE species IS NULL; 
```

Comment:
- [ ]  Kindly note that this query should return the number of visits from vets that do not specialize in that animal's species. Since Vet William Tatcher is specialized in Pokemon, Vet Stephanie Mendez is specialized in Digimon and Pokemon, Vet Jack Harkness is specialized in Digimon. This means that if Jack Harkness visits an animal from the Pokemon species, that animal should be counted too. Counting all will return 12, not 9 as displayed in your screenshot. Kindly fix this, please.
![image](https://user-images.githubusercontent.com/94551484/229279036-5b3eeeb3-c556-4e94-88d7-31550da4f988.png)